contracts/SpokePool.sol

```diff
@@ -948,20 +1025,28 @@ abstract contract SpokePool is
             // recipient wants wrappedNativeToken, then we can assume that wrappedNativeToken is already in the
             // contract, otherwise we'll need the user to send wrappedNativeToken to this contract. Regardless, we'll
             // need to unwrap it to native token before sending to the user.
-            if (!useContractFunds)
+            if (!relayExecution.slowFill)
                 IERC20Upgradeable(relayData.destinationToken).safeTransferFrom(msg.sender, address(this), amountToSend);
             _unwrapwrappedNativeTokenTo(payable(relayData.recipient), amountToSend);
             // Else, this is a normal ERC20 token. Send to recipient.
         } else {
             // Note: Similar to note above, send token directly from the contract to the user in the slow relay case.
-            if (!useContractFunds)
+            if (!relayExecution.slowFill)
                 IERC20Upgradeable(relayData.destinationToken).safeTransferFrom(
                     msg.sender,
                     relayData.recipient,
                     amountToSend
                 );
             else IERC20Upgradeable(relayData.destinationToken).safeTransfer(relayData.recipient, amountToSend);
         }
+
+        if (Address.isContract(relayData.recipient) && relayData.message.length > 0) {
+            AcrossMessageHandler(relayData.recipient).handleAcrossMessage(
```

Comment:
So if this fails, then relay is not filled at all and depositor can update the recipient/data, right?

Reply:
Exactly!